Testing on the snap-jolt project uncovered one minor issue:  if the application doesn't explicitly depend on OSHI, then it can crash at runtime with
```console
Error: Exception in thread "main" java.lang.NoClassDefFoundError: oshi/SystemInfo
> Task :PrintConfig FAILED
	at electrostatic4j.snaploader.platform.util.NativeVariant$Cpu.readFeatureFlags(NativeVariant.java:280)
2 actionable tasks: 2 executed
	at electrostatic4j.snaploader.platform.util.NativeVariant$Cpu.hasExtensions(NativeVariant.java:350)
	at electrostatic4j.snaploader.platform.util.PlatformPredicate.<init>(PlatformPredicate.java:147)
	at com.github.stephengold.snapjolt.PrintConfig.main(PrintConfig.java:52)
Caused by: java.lang.ClassNotFoundException: oshi.SystemInfo
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
	... 4 more
```

Adding an explicit dependency to the buildscript solves the problem:

https://github.com/stephengold/snap-jolt/commit/d88d58bb9d635d0503623ab69930d0750791fee0

but it would be better if jSnapLoader simply included a transitive dependency on OSHI.

--- a/snaploader-examples/build.gradle
+++ b/snaploader-examples/build.gradle
@@ -26,6 +26,12 @@ tasks.register("TestBasicFeatures2") {
     application.mainClass = 'electrostatic4j.snaploader.examples.TestBasicFeatures2'
 }
 
+tasks.register("TestCpuFeatures", JavaExec) {
+    classpath sourceSets.main.runtimeClasspath
+    description = 'Runs the TestCpuFeatures example app.'
+    mainClass = 'electrostatic4j.snaploader.examples.TestCpuFeatures'
+}
+
 tasks.register("MonitorableExample") {
     application.mainClass = 'electrostatic4j.snaploader.examples.MonitorableExample'
 }

--- a/snaploader-examples/build.gradle
+++ b/snaploader-examples/build.gradle
@@ -86,4 +86,11 @@ task createJar(type : Jar, dependsOn : copyLibs){
 
 dependencies {
     implementation project(path: ':snaploader')
+    implementation 'org.apache.logging.log4j:log4j-slf4j2-impl:2.24.3'
+
+    implementation 'com.github.stephengold:jolt-jni-Linux64:0.9.7'
+    runtimeOnly 'com.github.stephengold:jolt-jni-Linux64:0.9.7:DebugSp'
+    runtimeOnly 'com.github.stephengold:jolt-jni-Linux64_fma:0.9.7:DebugSp'
+    runtimeOnly 'com.github.stephengold:jolt-jni-Windows64:0.9.7:DebugSp'
+    runtimeOnly 'com.github.stephengold:jolt-jni-Windows64_avx2:0.9.7:DebugSp'
 }

--- a/snaploader-examples/src/main/java/electrostatic4j/snaploader/examples/TestCpuFeatures.java
+++ b/snaploader-examples/src/main/java/electrostatic4j/snaploader/examples/TestCpuFeatures.java
@@ -1,0 +1,111 @@
+/*
+ * Copyright (c) 2025, The Electrostatic-Sandbox Distributed Simulation Framework, jSnapLoader
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'AvrSandbox' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package electrostatic4j.snaploader.examples;
+
+import com.github.stephengold.joltjni.Jolt;
+import electrostatic4j.snaploader.LibraryInfo;
+import electrostatic4j.snaploader.LoadingCriterion;
+import electrostatic4j.snaploader.NativeBinaryLoader;
+import electrostatic4j.snaploader.filesystem.DirectoryPath;
+import electrostatic4j.snaploader.platform.NativeDynamicLibrary;
+import electrostatic4j.snaploader.platform.util.NativeVariant;
+import electrostatic4j.snaploader.platform.util.PlatformPredicate;
+
+/**
+ * Tests selection between native libraries based on CPU features.
+ *
+ * @author Stephen Gold sgold@sonic.net
+ */
+public final class TestCpuFeatures {
+
+    public static void main(String[] argv) {
+        // Test for each of the relevant CPU features:
+        System.out.println("avx    = " + NativeVariant.Cpu.hasExtensions("avx"));
+        System.out.println("avx2   = " + NativeVariant.Cpu.hasExtensions("avx2"));
+        System.out.println("bmi1   = " + NativeVariant.Cpu.hasExtensions("bmi1"));
+        System.out.println("f16c   = " + NativeVariant.Cpu.hasExtensions("f16c"));
+        System.out.println("fma    = " + NativeVariant.Cpu.hasExtensions("fma"));
+        System.out.println("sse4_1 = " + NativeVariant.Cpu.hasExtensions("sse4_1"));
+        System.out.println("sse4_2 = " + NativeVariant.Cpu.hasExtensions("sse4_2"));
+
+        // Define a custom predicate for Linux with all 7 CPU features:
+        PlatformPredicate linuxWithFma = new PlatformPredicate(
+                PlatformPredicate.LINUX_X86_64,
+                "avx", "avx2", "bmi1", "f16c", "fma", "sse4_1", "sse4_2");
+        System.out.println("linuxWithFma    = " + linuxWithFma.evaluatePredicate());
+
+        // Define a custom predicate for Windows with 4 CPU features:
+        PlatformPredicate windowsWithAvx2 = new PlatformPredicate(
+                PlatformPredicate.WIN_X86_64,
+                "avx", "avx2", "sse4_1", "sse4_2");
+        System.out.println("windowsWithAvx2 = " + windowsWithAvx2.evaluatePredicate());
+        System.out.flush();
+
+        LibraryInfo info = new LibraryInfo(
+                new DirectoryPath("linux/x86-64/com/github/stephengold"),
+                "joltjni", DirectoryPath.USER_DIR);
+        NativeBinaryLoader loader = new NativeBinaryLoader(info);
+        NativeDynamicLibrary[] libraries = {
+            new NativeDynamicLibrary("linux/x86-64-fma/com/github/stephengold", linuxWithFma), // must precede vanilla LINUX_X86_64
+            new NativeDynamicLibrary("linux/x86-64/com/github/stephengold", PlatformPredicate.LINUX_X86_64),
+            new NativeDynamicLibrary("windows/x86-64-avx2/com/github/stephengold", windowsWithAvx2), // must precede vanilla WIN_X86_64
+            new NativeDynamicLibrary("windows/x86-64/com/github/stephengold", PlatformPredicate.WIN_X86_64)
+        };
+        loader.registerNativeLibraries(libraries).initPlatformLibrary();
+        loader.setLoggingEnabled(true);
+        loader.setRetryWithCleanExtraction(true);
+        try {
+            loader.loadLibrary(LoadingCriterion.INCREMENTAL_LOADING);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed to load the joltjni library!");
+        }
+        System.err.flush();
+
+        // Invoke native code to obtain the configuration of the native library.
+        String configuration = Jolt.getConfigurationString();
+        /*
+         * Depending which native library was loaded, the configuration string
+         * should be one of the following:
+         *
+         * On LINUX_X86_64 platforms, either
+         *  Single precision x86 64-bit with instructions: SSE2 SSE4.1 SSE4.2 AVX AVX2 F16C LZCNT TZCNT FMADD (Debug Renderer) (16-bit ObjectLayer) (Assertions) (ObjectStream) (Debug) (C++ RTTI) (C++ Exceptions)
+         * or
+         *  Single precision x86 64-bit with instructions: SSE2 (Debug Renderer) (16-bit ObjectLayer) (Assertions) (ObjectStream) (Debug) (C++ RTTI) (C++ Exceptions)
+         *
+         * On WIN_X86_64 platforms, either
+         *  Single precision x86 64-bit with instructions: SSE2 SSE4.1 SSE4.2 AVX AVX2 F16C LZCNT TZCNT (FP Exceptions) (Debug Renderer) (16-bit ObjectLayer) (Assertions) (ObjectStream) (Debug) (C++ RTTI) (C++ Exceptions)
+         * or
+         *  Single precision x86 64-bit with instructions: SSE2 (FP Exceptions) (Debug Renderer) (16-bit ObjectLayer) (Assertions) (ObjectStream) (Debug) (C++ RTTI) (C++ Exceptions)
+         */
+        System.out.println(configuration);
+    }
+}

--- a/snaploader/build.gradle
+++ b/snaploader/build.gradle
@@ -27,5 +27,5 @@ jar { // assemble jar options [java -jar]
 }
 
 dependencies {
-    implementation('com.github.oshi:oshi-core:6.7.0')
+    api('com.github.oshi:oshi-core:6.7.0')
 }

